dask/dataframe/io/csv.py

```diff
@@ -724,51 +742,62 @@ def to_csv(
     -------
     The names of the file written if they were computed right away
     If not, the delayed tasks associated to the writing of the files
-    """
-    if PY2:
-        default_encoding = None
-        mode = "wb"
-    else:
-        default_encoding = "utf-8"
-        mode = "wt"
-
-    encoding = kwargs.get("encoding", default_encoding)
 
-    files = open_files(
-        filename,
+    Raises
+    ------
+    ValueError
+        If `header_first_partition_only` is set to `False` or
+        `name_function` is specified when `single_file` is `True`.
+    """
+    if single_file and name_function is not None:
+        raise ValueError("name_function is not supported under the single file mode")
+    if header_first_partition_only is None:
```

Comment:
Thanks, this looks good. Could you add tests that hit the new behavior (I think just the `elif not on L756`).